sandbox/sandbox.go

```diff
@@ -478,29 +479,28 @@ func createSingleSandbox(sandboxDef SandboxDef) (err error, execList []concurren
 			return sbError("check directory", "%s", err), emptyExecutionList
 		}
 	}
+
 	logger.Printf("Checking port %d using CheckPort\n", sandboxDef.Port)
-	err = CheckPort("createSingleSandbox", sandboxDef.SBType, sandboxDef.InstalledPorts, sandboxDef.Port)
-	if err != nil {
-		return sbError("check port", "%s", err), emptyExecutionList
+	if err := CheckPort("CreateSingleSandbox", sandboxDef.SBType, sandboxDef.InstalledPorts, sandboxDef.Port); err != nil {
+		return nil, err
 	}
 
-	err = os.Mkdir(sandboxDir, 0755)
-	if err != nil {
-		return sbError("sandbox dir creation", "%s", err), emptyExecutionList
+	if err := os.MkdirAll(sandboxDir, os.ModePerm); err != nil {
+		return nil, errors.Wrapf(err, "cannot create sandbodx dir %s", sandboxDir)
```

Comment:
This is an interesting suggestion. I should make some experiments to see how the user experience is going to be affected.